README.md

```diff
@@ -78,6 +107,106 @@ Properties:
+| scope   |       `:local` or `:default`        |    No    | `:local`                                         | Scope of the configuration to modify. In most circumstances, you should *not* change this.                                                                                                   |
+| config  |                Hash                 | **Yes**  |                                                  | Configuration to apply to the .conf file. This hash is structured as follows: `{ stanza: { key: 'value' } }`. See below for more detailed explanation of the config property.                |
+| user    |            String or nil            |    No    | Owner of the current Splunk installation, if any | User that will be used to write to the .conf files.                                                                                                                                          |
+| reset   |            true or false            |    No    | false                                            | When specified as true, entirely replaces existing config. By default, config is merged into the existing conf file.                                                                         |
+
+##### Configuration
+The configuration you provide will be evaluated against the existing config file (if it exists, and if reset is not specified).
+Symbol or string keys are irrelevant as the provided config is converted to string keys and values before being used.
+
+Some things to be aware of when using the configuration resource:
+
+###### **"Global" properties are evaluated under `[default]`**
+Properties defined at the top of the .conf file and not considered to be part of a stanza are assumed to be `[default]`
+per Splunk documentation: http://docs.splunk.com/Documentation/Splunk/6.4.2/Admin/Propsconf#GLOBAL_SETTINGS
+
+###### **Comments are not preserved**
+The configuration resource can't evaluate and merge the configuration on disk and by Chef while keeping the
+relevant comments and whitespace. The .conf file will be rewritten with the merged config and uniform whitespace.
+
+###### **Example of how the configuration is written out to disk:**
+
+```Ruby
+splunk_conf 'system/test.conf' do
+  config(
+    testing: {
+      one: 1,
+      two: 2,
+      three: 3
+    },
+    stanza: {
+      key: :value
+    }
+  )
+end
+```
+
+`/opt/splunk/etc/system/local/test.conf`:
+```INI
+# Warning: This file is managed by Chef!
+# Comments will not be preserved and configuration may be overwritten.
+
+[testing]
+one = 1
+two = 2
+three = 3
+
+[stanza]
+key = value
+```
+
+##### Run State
+Since there is no concept of type in .conf files, everything is evaluated as strings. Configuration is read as strings
+and the provided config is converted to strings before merging. `splunk_conf` will store config (after merging) into the
```

Comment:
It's another one of those things that we decided would be nice to have in the run state for consumers of the resource, though I don't know of any use case for it. It was trivial to implement, but I can just as easily remove it if we don't think we'll need it. One could instead use the read_config helper method to get the actual current configuration in a file.